src/winforms/toga_winforms/colors.py

```diff
@@ -1,12 +1,11 @@
 from .libs import Color
 
-CACHE = {}
+CACHE = {None: Color.Empty}
 
 
 def native_color(c):
-    try:
-        color = CACHE[c]
-    except KeyError:
-        color = Color.FromArgb(int(c.rgba.a * 255), c.rgba.r, c.rgba.g, c.rgba.b)
-        CACHE[c] = color
+    if c in CACHE:
+        return CACHE[c]
+    color = Color.FromArgb(int(c.rgba.a * 255), c.rgba.r, c.rgba.g, c.rgba.b)
+    CACHE[c] = color
```

Comment:
The try/catch lookup is a deliberate pattern; it's slightly faster, because the "success" path only needs to look up the key once, whereas `if c in cache; return cache[c]` requires two identical lookups of the same key.

Reply:
according to [this](https://softwareengineering.stackexchange.com/questions/225238/python-if-foo-in-dict-vs-try-dictfoo) both options are not optimal. using the third option suggested with the `dict.get` method.

Reply:
That post isn't an especially good source; there's a few people weighing in with opinions that have very clearly been taught Java, and are bringing Java language into the discussion.

Also, the get approach requires an extra comparison. It's not a huge deal; but idiomatic Python doesn't require the comparison. I'll make the tweak (along with an unrelated problem with colors in Cocoa that this new demo revealed) and merge.